paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -396,6 +434,86 @@ def sync_secrets(
     return True
 
 
+def sync_database_credentials(
+    kube_client: KubeClient,
+    cluster: str,
+    service: str,
+    secret_provider_name: str,
+    vault_cluster_config: Dict[str, str],
+    soa_dir: str,
+    vault_token_file: str,
+) -> bool:
+    config_loader = PaastaServiceConfigLoader(service=service, soa_dir=soa_dir)
+    system_paasta_config = load_system_paasta_config()
+    database_credentials_vault_overrides = (
+        system_paasta_config.get_database_credentials_vault_overrides()
+    )
+
+    for instance_config in config_loader.instance_configs(
+        cluster=cluster, instance_type_class=KubernetesDeploymentConfig
+    ):
+        database_credentials = instance_config.get_database_credentials()
+        with set_env(**database_credentials_vault_overrides):
+            # expects VAULT_ADDR_OVERRIDE, VAULT_CA_OVERRIDE, and VAULT_TOKEN_OVERRIDE to be set
+            # in order to use a custom vault shard. overriden temporarily in this context
+            provider = get_secret_provider(
+                secret_provider_name=secret_provider_name,
+                soa_dir=soa_dir,
+                service_name=service,
+                cluster_names=[cluster],
+                # overridden by env variables but still needed here for spec validation
+                secret_provider_kwargs={
+                    "vault_cluster_config": vault_cluster_config,
+                    "vault_auth_method": "token",
+                    "vault_token_file": vault_token_file,
+                },
+            )
+
+        for dstore, credentials in database_credentials.items():
+            for credential in credentials:
+                vault_path = f"secrets/database/{dstore}/{credential}"
+                secrets = provider.get_data_from_vault_path(vault_path)
+                if not secrets:
```

Comment:
@nemacysts if I understand correctly, this should be changed to "if secrets is None", thus allowing "secrets = {}" to be synced to K8s?

Reply:
depends on what y'all are expecting - if for the DRE case a secret not existing and a secret being empty (or I guess malformed if empty is not a valid value) are the same, then we can leave this line as-is :)

Reply:
ack - if we're not expecting the vault process that's manipulating this data to ever get us into this scenario, then that seems fine to me

(if we're not sure, it'd be nice to add the check to avoid a page on our end :))